qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -465,50 +467,172 @@ def _has_unitary_(self):
+
+        # Complete the addition in order to get the sign bit of (a' + b).
+        ancillas[-1], a_split[0] = bb.add(CNOT(), ctrl=ancillas[-1], target=a_split[0])
+        b_split[0], a_split[0] = bb.add(CNOT(), ctrl=b_split[0], target=a_split[0])
+
+        # Use a 0-controlled NOT gate in order to flip the target bit if the sign bit of (b' + a)'
+        # is 1. (b' + a)' = b - a therefore if a > b, then b - a < 0 and the sign bit of b - a will
+        # be 1.
+        a_split[0] = bb.add(XGate(), q=a_split[0])
+        a_split[0], target = bb.add(CNOT(), ctrl=a_split[0], target=target)
+        a_split[0] = bb.add(XGate(), q=a_split[0])
+
+        # Uncompute the completion of addition on the last bit of a.
+        b_split[0], a_split[0] = bb.add(CNOT(), ctrl=b_split[0], target=a_split[0])
+        ancillas[-1], a_split[0] = bb.add(CNOT(), ctrl=ancillas[-1], target=a_split[0])
+
+        # Iteratively uncomputes the left adder circuit building block by performing the operations
+        # in reverse order. In a normal adder circuit we would use the right adder circuit building
+        # block, but because we only need to compute the carry-out bit we uncompute the circuit to
+        # restore a and b.
+        for i in range(true_bitsize - 1):
+            and_ctrl = and_ctrls.pop()
+            ancilla = ancillas.pop()
+
+            if i < true_bitsize - 2:
+                carry_in = ancillas[-1]
+                carry_in, ancilla = bb.add(CNOT(), ctrl=carry_in, target=ancilla)
+
+            and_ctrl = bb.add(And(uncompute=True), ctrl=and_ctrl, target=ancilla)
+            b_split[i + 1] = and_ctrl[0]
+            a_split[i + 1] = and_ctrl[1]
+
+            if i < true_bitsize - 2:
+                carry_in, a_split[i + 1] = bb.add(CNOT(), ctrl=carry_in, target=a_split[i + 1])
+                ancillas[-1], b_split[i + 1] = bb.add(CNOT(), ctrl=carry_in, target=b_split[i + 1])
+
+        # Uncompute the bitflips done to represent x'.
+        for i in range(true_bitsize):
+            b_split[i] = bb.add(XGate(), q=b_split[i])
+
+        a = bb.join(a_split)
+        b = bb.join(b_split)
+
+        # If the input registers were unsigned we free the ancilla sign bits.
+        if not self.signed:
+            a_split = bb.split(a)
+            a_sign = a_split[0]
+            a = bb.join(a_split[1:])
+            bb.free(a_sign)
+
+            b_split = bb.split(b)
+            b_sign = b_split[0]
+            b = bb.join(b_split[1:])
+            bb.free(b_sign)
+
+        # Return the output registers.
+        return {'a': a, 'b': b, 'target': target}
 
-    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
-        # TODO Determine precise clifford count and/or ignore.
-        # See: https://github.com/quantumlib/Qualtran/issues/219
-        # See: https://github.com/quantumlib/Qualtran/issues/217
-        t_complexity = self.t_complexity()
-        return {(TGate(), t_complexity.t)}
```

Comment:
@mpharrigan I remember in an older comment you said the call graph should report the complexity in the most natural way possible or something like that. are we still following that?

here -and in all arithmatic operations- the natural way would be to report it in Toffoli count rather than T count. since that will affect the resource estimation result if CZ magic factories are used instead of T factories.

Reply:
The call graph "callees" should be as high-level as possible. The base arithmetic operations should report toffoli (probably) and the arithmetic operations that are built on top of other arithmetic operations should use those.

Reply:
Closing this since it is in reference to Log Depth GreaterThan()